extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -589,7 +585,20 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
       // otherwise, check if the hydrated segmentations are in the same FOR
       // as the primary displaySet, if so add the representation (since it was not there)
       const { id: segDisplaySetInstanceUID, type } = segmentation;
-      const segFrameOfReferenceUID = this._getFrameOfReferenceUID(segDisplaySetInstanceUID);
+      let segFrameOfReferenceUID = this._getFrameOfReferenceUID(segDisplaySetInstanceUID);
+
+      if (!segFrameOfReferenceUID) {
+        // if the segmentation does not have a FOR, we might check the
```

Comment:
It is confusing that the word segmentation appears twice in this sentence. Should this say instead, "if the segmentation display set does not..."?

Reply:
Also using FOR is confusing, I would stick with 'frame of reference uid' in the comment.